pkg/controllers/provisioning/v1alpha1/reallocation/controller.go

```diff
@@ -78,7 +79,9 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 	}
 
 	// 3. Get Nodes with Underutilized Label for resource usage reevaluation and Remove TTL if necessary
-	labeledUnderutilized, err := c.filter.GetLabeledUnderutilizedNodes(ctx, provisioner)
+	labeledUnderutilized, err := c.filter.GetLabeledNodes(ctx, provisioner, map[string]string{
```

Comment:
Consider encapsulating this inside of the filter function. Something like filter.GetUnderutilizedNodes() seems idiomatic to me. Looking at this more closely, though, How does this filter differ from line 71?

Reply:
I decided to create the `GetLabeledNodes` function because I saw that the filter file was getting extremely long with repeated logic, and wanted to simplify it all. 

`GetLabeledNodes` simply gets nodes with the associated labels of the provisioner and any additional specified labels. 

`GetUnderutilizedNodes` and  `GetTerminableNodes` both require special logic (checking if it is underutilized and checking if is past TTL.

Reply:
`GetCordonableNodes` and `GetTTLableNodes` have special cases as well that require a simple check as opposed to a label check.